docs/data/data-grid/clipboard/clipboard.md

```diff
@@ -0,0 +1,117 @@
+The paste operation only affects cells in the columns that are [`editable`](/x/react-data-grid/editing/#making-a-column-editable).
+
+Same as with editing, you can use `valueParser` to modify the pasted value and `valueSetter` to update the row with new values.
+See [Value parser and value setter](/x/react-data-grid/editing/#value-parser-and-value-setter) section of the editing documentation for more details.
+
+The behavior of the clipboard paste operation depends on the selection state of the data grid and the data pasted from clipboard.
+The priority is the following, from highest to lowest:
+
+1. If multiple cells are selected (see [Cell selection<span class="plan-premium" title="Premium plan"></span>](/x/react-data-grid/cell-selection/)), the selected cells are updated with the pasted values.
+2. If one or more rows are selected (see [Row selection](/x/react-data-grid/row-selection/)), the selected rows are updated with the pasted values.
+3. If a single cell is selected, the values are pasted starting from the selected cell.
+
+{{"demo": "ClipboardPaste.js", "bg": "inline"}}
+
+### Disable clipboard paste
+
+To disable clipboard paste, set the `disableClipboardPaste` prop to `true`:
+
+{{"demo": "ClipboardPasteDisabled.js", "bg": "inline"}}
+
+### Persisting pasted data
+
+Clipboard paste uses the same API for persistence as [Editing](/x/react-data-grid/editing/#persistence)—use the `processRowUpdate` prop to persist the updated row in your data source:
+
+```tsx
+processRowUpdate?: (newRow: R, oldRow: R) => Promise<R> | R;
+```
+
+The row will be updated with a value returned by the `processRowUpdate` callback.
+If the callback throws or returns a rejected promise, the row will not be updated.
+
+The demo below shows how to persist the pasted data in the browser's `sessionStorage`.
+
+{{"demo": "ClipboardPastePersistence.js", "bg": "inline", "defaultCodeOpen": false}}
+
+### Events
+
+The following events are fired during the clipboard paste operation:
+
+- `clipboardPasteStart` - fired when the clipboard paste operation starts
+- `clipboardPasteEnd` - fired when all row updates from clipboard paste have been processed
+
+For convenience, you can also listen to these events using their respective props:
+
+- `onClipboardPasteStart`
+- `onClipboardPasteEnd`
+
+The demo below shows how to use these events to display a loading indicator while the clipboard paste operation is in progress:
+
+{{"demo": "ClipboardPasteEvents.js", "bg": "inline"}}
+
+## Format of the clipboard data
+
+By default, the clipboard copy and paste operations use the following format:
+
+- The cell values are separated by a tab (`\t`) character.
+- The rows are separated by a new line (`\n`) character.
+
+You can use `clipboardCopyTextDelimiter` and `unstable_splitClipboardPastedText` props to change the format:
```

Comment:
Oops, my bad, I suggested this change based on my other suggestion. The prop name is `clipboardCopyCellDelimiter`. Should we correct this before merging the release?
@cherniavskii